generic: Fix missing symbol with LXC options enabled

danielfdickinson opened the pull request:
At least when building with LXC options enabled this symbol
needs to be present or the kernel oldconfig hangs due to new
symbol.

Signed-off-by: Daniel F. Dickinson <cshored@thecshore.com>

This gets building and running working for me on the Raspberry Pi Model B+ (brcm2708/bcm2708).

hauke commented:
Please move this to target/linux/generic/config-4.14 and probably also target/linux/generic/config-4.19
I assume that this would affect more targets and when someone runs "make kernel_oldconfig CONFIG_TARGET=subtarget" on the target it will probably be removed.

Commit:
config: kernel: Fix missing symbol on brcm2708 with CGROUPS

When CGROUP block io is enabled a new symbol is exposed and needs to
be set or unset else kernel oldconfig hangs waiting for input during
normal OpenWrt builds.  Therefore add sane defaults for this symbol
in that case.  Also, the defaults brcm2708 are different than generic
defaults because the platform's defconfig enables BLK_DEV_THROTTLING
by default (in defconfig config from the patches used to match
upstream's kernel, not in OpenWrt config-4.xx).

Signed-off-by: Daniel F. Dickinson <cshored@thecshore.com>

cshoredaniel commented:
About to force-push a change, but please wait until I run test it (will ping) -- menuconfig works though (it turns out there is some related help text in the CGROUPS information but not actual KConfig symbols AND that brcm2708 is somewhat special about this (see commit message once force pushed)).

cshoredaniel commented:
Compilation succeeds on brcm2708 (rpi B+) - in a bit we'll see what does to the Pi...

cshoredaniel commented:
Run tested on Raspberry Pi B+ -- will probably take some time before I can get to other devices...

chunkeey commented:
> At least when building with LXC options enabled this symbol
needs to be present or the kernel oldconfig hangs due to new
symbol.

There's also a discussion going on the ML with a patch to bring LXC up to snuff.

https://patchwork.ozlabs.org/patch/1019300/

cshoredaniel commented:
I've posted on ML since that seems to be preferred: http://lists.infradead.org/pipermail/openwrt-devel/2019-January/015439.html
